### PR TITLE
🔥🔥🔥 WIP 🔥🔥🔥 - started adding custom subs instead of hard coded ones

### LINKED ISSUE #2: Non-Static Subs
Users should be able to create subs on their own (configurable) so either the admin/s or user/s can create new subs.

### PR DESCRIPTION
**Pull request by bdbch**
# THIS PULL REQUEST IS IN WIP STATE AND SHOULD NOT BE MERGED

closes #2 

---

This pull request **will** implement subs/categories which can be created and subscribed by the user.

## Todo
* [x] Create models in server
* [x] Create routes in server
* [ ] Create validation and creation process in server
* [ ] Add all correct category actions to redux
* [ ] Hook up all frontend components/containers with new state
* [ ] Show *not found* page on /a/ route whenever the requested category does not exist
* [ ] Create category screen + process
* [ ] Show subreddit/category owner on category page
* [ ] Show subreddits/categories on user page

## Notable changes
* Updated React and ReactDOM to 16.8.0 so we can use hooks and effects (maybe stop using stateful components?)

There may be more. Check it out and help me with it if you want to. I need some rest. It's a lot of fun!

### CODE REVIEW

File: client/src/components/Sidebar/Component.js [hunk 1]
```diff
@@ -11,17 +11,23 @@ const Wrapper = styled.aside`
...
-const Sidebar = ({ token }) => (
-  <Wrapper>
-    {token && <SidebarCreatePostButton />}
-    <SidebarCategoryList />
-  </Wrapper>
-);
+const Sidebar = props => {
+  useEffect(() => {
+    props.fetchCategories();
+  }, []);
```
Comment (bdbch) [inline client/src/components/Sidebar/Component.js]: I'll need to revisit here. I think we will fetch the users *subscribed* categories here, right? Listing all categories won't make sense anyway (maybe make them accessible via search later).

File: client/package.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
   "dependencies": {
     "jwt-decode": "^2.2.0",
     "moment": "^2.23.0",
-    "react": "^16.7.0",
-    "react-dom": "^16.7.0",
+    "react": "^16.8.0",
```
Comment (bdbch) [inline client/package.json]: react magic 🧙‍♂️

File: client/src/components/Sidebar/Component.js (hunk 1 above)
Comment (d11z) [reply client/src/components/Sidebar/Component.js]: Yes, I still haven't decided how I want that to go exactly. Perhaps we add 'front page' and 'popular' pages like Reddit for discovering new ones. Also I was thinking we should rename categories into communities maybe.

Comment (bdbch) [reply client/src/components/Sidebar/Component.js]: I'd go for front page and popular. "all" already sounds strange. updated it in this pull request already. Communities sounds good!

**Pull request closed without merging**